api/src/services/payload.ts

```diff
@@ -261,6 +261,15 @@ export class PayloadService {
 	processDates(payloads: Partial<Record<string, any>>[], action: Action): Partial<Record<string, any>>[] {
 		const fieldsInCollection = Object.entries(this.schema.collections[this.collection].fields);
 
+		// Update specific database type overrides
+		for (const [_name, field] of fieldsInCollection) {
+			if (field.special.includes('cast-timestamp')) {
+				field.type = 'timestamp';
+			} else if (field.special.includes('cast-datetime')) {
+				field.type = 'dateTime';
+			}
+		}
```

Comment:
I'm wondering if this should happen further upstream. I believe there's already a function in get-schema that parses the type:

https://github.com/directus/directus/blob/e2eb2801c5884b5170af93c0cbb715aae11b80d7/api/src/utils/get-schema.ts#L134

so these cast flags should be instead of in the methods:

https://github.com/directus/directus/blob/e2eb2801c5884b5170af93c0cbb715aae11b80d7/api/src/utils/get-local-type.ts#L117-L125

Reply:
Agreed, it's better to be done upstream as the field type will be correct within the schema. It's clearer to be alongside other typecasting field flags too. Thanks @rijkvanzanten! 🙌🏼